tests/test_covar2d.py

```diff
@@ -154,3 +156,59 @@ def testGetCWFCoeffsClean(self):
         self.assertTrue(
             np.allclose(results, self.coeff_cwf_clean, atol=utest_tolerance(self.dtype))
         )
+
+    def testGetCWFCoeffsCleanCTF(self):
+        """
+        Test case of clean images (coeff_clean and noise_var=0)
+        while using a non Identity CTF.
+
+        This case may come up when a developer switches between
+        clean and dirty images.
+        """
+
+        coeff_cwf = self.cov2d.get_cwf_coeffs(
+            self.coeff_clean, self.h_ctf_fb, self.h_idx, noise_var=0
+        )
+
+        # Reconstruct images from output of get_cwf_coeffs
+        img_est = self.basis.evaluate(coeff_cwf)
+        # Compare with clean images
+        delta = np.mean(np.square((self.imgs_clean - img_est).asnumpy()))
+        self.assertTrue(delta < 0.02)
+
+    def testGetCWFCoeffsCTFargs(self):
+        """
+        Test we raise when user supplies incorrect CTF arguments
+        """
+
+        with raises(RuntimeError):
+            _ = self.cov2d.get_cwf_coeffs(
+                self.coeff, self.h_ctf_fb, None, noise_var=self.noise_var
+            )
+
+    @parameterized.expand(
```

Comment:
`@pytest.mark.parametrize` and save us the dependency?

Reply:
AFAIK that doesn't work with unittest TestCase class... I can rewrite the tests to be functional instead of class based, but then we'll have two different styles of writing unit tests...

Reply:
Ah that's too bad. Good to know.